src/highdicom/pr/sop.py

```diff
@@ -0,0 +1,1459 @@
+            **kwargs
+        )
+
+        # Softcopy Presentation LUT
+        if presentation_luts is not None:
+            if presentation_lut_shape is not None:
+                raise TypeError(
+                    'Only one of "presentation_luts" or '
+                    '"presentation_lut_shape" should be provided.'
+                )
+            if len(presentation_luts) == 0:
+                raise ValueError(
+                    'Argument "presentation_luts" must not be empty.'
+                )
+            for v in presentation_luts:
+                if not isinstance(v, LUT):
+                    raise TypeError(
+                        'Items of "presentation_luts" should be of type '
+                        'highdicom.LUT.'
+                    )
+            self.PresentationLUTSequence = presentation_luts
+        else:
+            presentation_lut_shape = (
+                presentation_lut_shape or
+                PresentationLUTShapeValues.IDENTITY
+            )
+            self.PresentationLUTShape = PresentationLUTShapeValues(
+                presentation_lut_shape
+            ).value
+
+
+class PseudoColorSoftcopyPresentationState(_SoftcopyPresentationState):
+
+    """SOP class for a Pseudo-Color Softcopy Presentation State object.
+
+    A Pseudo-Color Softcopy Presentation State object includes instructions for
+    the presentation of a grayscale image as a color image by software.
+
+    """
+
+    def __init__(
+        self,
+        referenced_images: Sequence[Dataset],
+        series_instance_uid: str,
+        series_number: int,
+        sop_instance_uid: str,
+        instance_number: int,
+        manufacturer: str,
+        manufacturer_model_name: str,
+        software_versions: Union[str, Tuple[str]],
+        device_serial_number: str,
+        content_label: str,
+        palette_color_lut: PaletteColorLookupTable,
+        content_description: Optional[str] = None,
+        graphic_annotations: Optional[Sequence[GraphicAnnotation]] = None,
+        graphic_layers: Optional[Sequence[GraphicLayer]] = None,
+        graphic_groups: Optional[Sequence[GraphicGroup]] = None,
+        concept_name_code: Union[Code, CodedConcept, None] = None,
+        institution_name: Optional[str] = None,
+        institutional_department_name: Optional[str] = None,
+        content_creator_name: Optional[Union[str, PersonName]] = None,
+        content_creator_identification: Optional[
+            ContentCreatorIdentificationCodeSequence
+        ] = None,
+        rescale_intercept: Union[int, float, None] = None,
+        rescale_slope: Union[int, float, None] = None,
+        rescale_type: Union[RescaleTypeValues, str, None] = None,
+        modality_lut: Optional[ModalityLUT] = None,
+        copy_modality_lut: bool = False,
+        softcopy_voi_luts: Optional[Sequence[SoftcopyVOILUT]] = None,
+        copy_voi_lut: bool = False,
+        icc_profile: Optional[ImageCmsProfile] = None,
```

Comment:
```suggestion
        icc_profile: Optional[bytes] = None,
```

Reply:
Internally, we should use [Pil.ImageCms.CmsProfile](https://pillow.readthedocs.io/en/stable/reference/ImageCms.html#PIL.ImageCms.CmsProfile) and we should perform some checks to ensure the [constraints imposed by the standard](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.11.15.html) are met:
- `device_class`
- `connection_space`
- `xcolor_space`

Reply:
Further, the ICC Profile is mandatory. Therefore, I would suggest including an sRGB profile if `icc_profile is None`:

```python
from PIL import ImageCms

if icc_profile is None:
    icc_profile = ImageCms.CmsProfile('sRGB')
```

Reply:
Added these checks in 874f6b8 

Now I can't figure out how to actually create profile that passes these checks, so tests are failing. Think I'm going to need your help generally on this ICC profile stuff.

Reply:
Oh boy...this is more complicated that I thought/hoped.

The Pillow library uses the [Little-CMS](https://github.com/mm2/Little-CMS) C library under the hood for color management. 
The [PIL.ImageCms.createProfile()](https://pillow.readthedocs.io/en/stable/reference/ImageCms.html#PIL.ImageCms.createProfile) function internally calls the C function [cmsCreate_sRGBProfile()](https://github.com/mm2/Little-CMS/blob/7cc53b4e9ab39e37da69955fe059f4b47514b41a/src/cmsvirt.c#L643), which calls [cmsCreateRGBProfileTHR](https://github.com/mm2/Little-CMS/blob/7cc53b4e9ab39e37da69955fe059f4b47514b41a/src/cmsvirt.c#L101). That function hard codes the device class to Display device (`"mntr"`) (see [here](https://github.com/mm2/Little-CMS/blob/7cc53b4e9ab39e37da69955fe059f4b47514b41a/src/cmsvirt.c#L119)), while we need an Input device (`"scnr"`).

Reply:
We will need a class or function to create a default **Input** device profile for the sRGB color space and Lab profile connection space. Maybe something we should contribute to Pillow?